docs/reference/ml/apis/find-file-structure.asciidoc

```diff
@@ -1269,9 +1276,405 @@ If the request does not encounter errors, you receive the following result:
 <8> `joda_timestamp_formats` are used to tell Logstash and Ingest pipeline how
     to parse timestamps.
 <9> `java_timestamp_formats` are the Java time formats recognized in the time
-    fields.  In future Ingest pipeline will switch to use this format.
+    fields. In future Ingest pipeline will switch to use this format.
 <10> The timestamp format in this sample doesn't specify a timezone, so to
      accurately convert them to UTC timestamps to store in Elasticsearch it's
      necessary to supply the timezone they relate to. `need_client_timezone`
      will be `false` for timestamp formats that include the timezone.
 
+If you try to analyze a lot of data then the analysis will take a long time.
+If you want to limit the amount of processing your {es} cluster performs for
+a request, use the timeout query parameter. The analysis will be aborted and
+an error returned when the timeout expires. For example, you can replace 20000
+lines in the previous example with 200000 and set a 1 second timeout on the
+analysis:
+
+[source,js]
+----
+curl -s "s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2018-06.csv" | head -200000 | curl -s -H "Content-Type: application/json" -XPOST "localhost:9200/_xpack/ml/find_file_structure?pretty&lines_to_sample=200000&timeout=1s" -T -
+----
+// NOTCONSOLE
+// Not converting to console because this shows how curl can be used
+
+Unless you are using an incredibly fast computer you'll receive a timeout error:
+
+[source,js]
+----
+{
+  "error" : {
+    "root_cause" : [
+      {
+        "type" : "timeout_exception",
+        "reason" : "Aborting structure analysis during [delimited record parsing] as it has taken longer than the timeout of [1s]"
+      }
+    ],
+    "type" : "timeout_exception",
+    "reason" : "Aborting structure analysis during [delimited record parsing] as it has taken longer than the timeout of [1s]"
+  },
+  "status" : 500
```

Comment:
`TransportCloseJobAction` throws `IllegalStateException` on timeout and `TransportOpenJobAction` throws `ElasticsearchException`.  Both these map to status 500.

For an example outside of X-Pack, `TransportRestoreSnapshotAction` throws `ProcessClusterEventTimeoutException`, and this maps to 503.  But the description of 503 says it's for temporary conditions which will be alleviated after some delay, so I'm not sure that's right for the file structure finder, because if an analysis is too time consuming first time it's likely to stay that way (at least for the same hardware).

Also, whoever originally wrote the `ElasticsearchTimeoutException` class had the opportunity to specify the most appropriate status for it and chose 500.

So although 500 seems suboptimal, it's probably the best there is and most consistent with the rest of ES.